console/src/tasks.rs

```diff
@@ -114,31 +138,67 @@ impl State {
         if let Some(now) = update.now {
             self.last_updated_at = Some(now.into());
         }
+
+        if let Some(new_metadata) = update.new_metadata {
+            let metas = new_metadata.metadata.into_iter().map(|meta| {
+                let id = meta.id.expect("no id").id;
+                let metadata = meta.metadata.expect("no metadata");
+                (id, metadata.into())
+            });
+            self.metas.extend(metas);
+        }
+
         let mut stats_update = update.stats_update;
         let sorted = &mut self.sorted_tasks;
-        let new_tasks = update.new_tasks.into_iter().filter_map(|task| {
+        let metas = &mut self.metas;
+        let new_tasks = update.new_tasks.into_iter().filter_map(|mut task| {
             if task.id.is_none() {
                 tracing::warn!(?task, "skipping task with no id");
             }
             let kind = match task.kind() {
                 proto::tasks::task::Kind::Spawn => "T",
                 proto::tasks::task::Kind::Blocking => "B",
             };
-            let fields = task
+
+            let fields: Vec<Field> = task
                 .fields
-                .iter()
-                .fold(String::new(), |mut res, f| {
-                    write!(&mut res, "{} ", f).unwrap();
-                    res
+                .drain(..)
+                .filter_map(|f| {
+                    let field_name = f.name.as_ref().expect("no name");
+                    let name: Option<Arc<str>> = match field_name {
+                        proto::field::Name::StrName(n) => Some(n.clone().into()),
+                        proto::field::Name::NameIdx(idx) => {
+                            let meta_id = f.metadata_id.as_ref().expect("no id");
```

Comment:
similarly, it might be more robust to bail if we can't find the metadata
```suggestion
                            let meta_id = f.metadata_id.as_ref()?;
```
(though we should probably log a warning saying that the subscriber sent us nonsensical data)